src/arrayMethodJoin.js

```diff
@@ -4,9 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if ((i === (this.length) - 1) && this[i] === undefined) {
+        str += '';
+      } else if (i === (this.length) - 1) {
+        str += this[i];
+      } else if (this[i] === undefined || this[i] === null) {
+        str += separator;
+      } else {
+        str += this[i] + String(separator);
+      }
```

Comment:
avoid such nested `if else` conditions
This code is hard to read and maintain

Reply:
Ok